storage_service/locations/models/gpg.py

```diff
@@ -251,28 +229,51 @@ def _gpg_encrypt(path, key_fingerprint):
     ``ok`` and ``status`` attributes, see
     https://pythonhosted.org/python-gnupg/.
     """
+    tar_created = False
     if os.path.isdir(path):
         _create_tar(path)
+        tar_created = True
     encr_path, result = gpgutils.gpg_encrypt_file(path, key_fingerprint)
-    if os.path.isfile(encr_path):
+    if os.path.isfile(encr_path) and result.ok:
         LOGGER.info('Successfully encrypted %s at %s', path, encr_path)
         os.remove(path)
         os.rename(encr_path, path)
         return path, result
     else:
+        if tar_created:
+            _extract_tar(path)
         fail_msg = _('An error occured when attempting to encrypt'
                      ' %(path)s' % {'path': path})
         LOGGER.error(fail_msg)
         raise GPGException(fail_msg)
 
 
+def _db_engine():
+    if 'sqlite' in settings.DATABASES['default']['ENGINE']:
+        return 'sqlite'
+    if 'mysql' in settings.DATABASES['default']['ENGINE']:
+        return 'mysql'
+    return 'other'
+
+
 def _encr_path2key_fingerprint(encr_path):
     """Given an encrypted path, return the fingerprint of the GPG key
     used to encrypt the package. Since it was already encrypted, its
     model must have a GPG fingerprint.
     """
-    return Package.objects.get(
-        current_path__endswith=encr_path).encryption_key_fingerprint
+    sql = ('SELECT * FROM locations_package WHERE %s LIKE CONCAT(\'%%\','
+           ' current_path, \'%%\')')
+    if _db_engine() == 'sqlite':
+        sql = ('SELECT * FROM locations_package WHERE %s LIKE "%" ||'
+               ' current_path || "%"')
```

Comment:
Couldn't we use [`contains`](https://docs.djangoproject.com/en/1.8/ref/models/querysets/#contains) instead of a raw query? That way we don't need to know the DB engine.

Reply:
I don't think this is possible @jraddaoui because we are not checking whether `current_path` contains a given string but whether a given string contains `current_path`. This is why I'm doing the db engine gymnastics. I'll think about this a bit more, but I don't think there is a better solution. (Please let me know if you see an obvious one.)

Reply:
Oh, right, it's the other way around. Would it be possible to strip the part we need from `encr_path`? If not, I can't find a better solution.

Reply:
Yeah, I thought of that, but I don't think there is a reliable way of stripping away the location's full path because we don't have access to the location instance. Also, we can't just use the filename of `encr_path` because it may not be a file within an AIP, which means it the `encr_path` file name will never be a substring of the package's current path.